refactor: signature autofilling for Simulate RPC

tequdev opened the pull request:
## High Level Overview of Change

Refactor to enable autofilling of signature-related fields in the Simulate RPC for use in the Signature field (like for Lending, Sponsor feature) as well

<!--
### Context of Change
-->

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

mvadari commented:
@tequdev is this PR ready to merge?

tequdev commented:
> @tequdev is this PR ready to merge?

yes👍

pratikmankawde commented on src/xrpld/rpc/handlers/Simulate.cpp:
```diff
@@ -72,39 +72,23 @@ getAutofillSequence(Json::Value const& tx_json, RPC::JsonContext& context)
+    if (sigObject.isMember(jss::Signers))
     {
-        if (!tx_json[jss::Signers].isArray())
+        if (!sigObject[jss::Signers].isArray())
```
Question/Suggestion: Shouldn't this check `!sigObject[jss::Signers].isArray()` happen even before this function is called?

bthomee replied:
@tequdev do you have a response?

tequdev replied:
@pratikmankawde What's the reason for that?

pratikmankawde replied:
The name of the function suggests that it fills signatures into objects. But this function is also checking if the json object passed-in has the sub-objects created with correct data types. I think this function should only be called if the object types are correct(Json is valid). Hence the check should happen before.

There should be a separate function to check the validity of the json. Separation of concerns and responsibilities.

Right now we return an error if the json object is malformed. At some point we may want to recreate the sub-objects of proper types.

mvadari replied:
That feels unnecessary - why look at the `jss:Signers` field twice in two different places? Makes way more sense to me to just check before you're going to need to work with it.

pratikmankawde replied:
My suggestion was to still do it once, just outside this function and before calling this. We then don't call this function if the field data types are not what we expect. Unless there's a case where `sigObject[jss::Signers]` is expected to be a scalar.

mvadari replied:
Instead of adding a precondition to this function that you have to remember to check before you call it, it feels more robust to check it inside the function. That's also generally the pattern in this repo.

bthomee committed:
refactor: Autofill signature for Simulate RPC

This change enables autofilling of signature-related fields in the Simulate RPC.
